src/output.cpp

```diff
@@ -419,17 +423,22 @@ std::string Output::map_hist_to_str(
     const std::vector<std::pair<std::vector<uint8_t>, uint64_t>>
         &total_counts_by_key) const
 {
-  std::vector<std::string> elems;
   uint32_t i = 0;
   const auto &map_info = bpftrace.resources.maps_info.at(map.name());
   const auto &map_type = map_info.value_type;
+  bool first = true;
   for (auto &key_count : total_counts_by_key) {
     auto &key = key_count.first;
     auto &value = values_by_key.at(key);
 
     if (top && values_by_key.size() > top && i++ < (values_by_key.size() - top))
       continue;
 
+    if (first)
+      first = false;
+    else
+      map_elem_delim(map_type);
```

Comment:
nit: can we just move this to the end of this loop and then call it again after the loop finishes? seems less ugly than keeping this `first` state var

Reply:
Sorry, I'm not understanding how that'd work. We don't want to print a trailing delimiter, so we need to emit fewer delimiters than map elements - how can we do that without a variable?